src/main/java/org/hawkular/client/android/backend/BackendClient.java

```diff
@@ -290,6 +287,11 @@ public void getTriggers(@NonNull Callback<List<Trigger>> callback) {
         getPipe(pipeName).read(getFilter(uri), callback);
     }
 
+    @SuppressWarnings("unchecked")
+    private <T> void savePipe(String pipeName, Object object, Callback<List<T>> callback) {
```

Comment:
Even if original `Pipe` interface uses `save` keyword, maybe `write` be more suitable. Also, and more important, you can change signature to `savePipe(String pipeName, T pipeItem, Callback<List<T>> callback)`.